decompiler/frontend/binaryninja/handlers/globals.py

```diff
@@ -12,30 +23,161 @@ class GlobalHandler(Handler):
+        value, type = self._get_unknown_value(variable.address, view, variable.address)
+        return GlobalVariable(
+                    name=self._lifter.lift(variable.symbol).name if variable.symbol else "data_" + f"{variable.address:x}",
+                    vartype=self._lifter.lift(type),
+                    ssa_label=parent.ssa_memory_version if parent else 0,
+                    initial_value=value
+                )
+
+
+    def _get_unknown_value(self, addr: int, view: BinaryView, caller_addr: int = 0):
+        """Return datavariable, string or raw bytes at given address."""
+        if datavariable := view.get_data_var_at(addr):
+            return self._lifter.lift(datavariable, view=view, caller_addr=caller_addr), datavariable.type
+        if (data := self._get_different_string_types_at(addr, view)) and data[0] is not None:
+            data, type = data[0], Type.pointer(view.arch, data[1])
         else:
-            return Constant(variable.value)
+            data, type = self._get_raw_bytes(addr, view), Type.pointer(view.arch, Type.void())
+
+        if len(data) > 129:
+            data = data[:129] + '..."'
+        return data, type
+            
+
+    def _get_raw_bytes(self, addr: int, view: BinaryView) -> str:
+        """Returns raw bytes as hex string after a given address to the next data structure or section"""
+        if (next_data_var := view.get_next_data_var_after(addr)) is not None:
+            data = view.read(addr, next_data_var.address - addr)
+        else:
+            data = view.read(addr, view.get_sections_at(addr)[0].end)
+
+        string = ''.join("\\x{:02x}".format(x) for x in data)
+        return f'"{string}"'
+
+
+    def _get_different_string_types_at(self, addr: int, view: BinaryView) -> Tuple[Optional[str], Type]:
+        """Extract string with char/wchar16/wchar32 type if there is one"""
+        types: list[Type] = [Type.char(), Type.wide_char(2), Type.wide_char(4)]
+        for type in types:
+            string = self._get_string_at(view, addr, type.width)
+            if string != None:
+                break
+        return string, type
+
+
+    def _get_string_at(self, view: BinaryView, addr: int, width: int) -> Optional[str]:
+        """Read string with specified width from location."""
+        raw_bytes: bytearray = bytearray()
+        match width:
+            case 1:
+                read = view.reader(addr).read8
+                identifier = ""
+            case 2:
+                read = view.reader(addr).read16
+                identifier = "L"
+            case 4:
+                read = view.reader(addr).read32
+                identifier = "L"
+            case _:
+                raise ValueError("Width not supported for reading bytes")
+
+        while (byte := read()) != 0x00:
+            if byte > 127: # ASCII (127) or UTF8 (255)?
+                return None
+            raw_bytes.append(byte)
+
+        string = str(raw_bytes)[12:-2]
```

Comment:
`raw_bytes` is a `bytearray` therefore the string representation is `bytearray(b'...')`. 
I simply purge the stuff around `...`. 
Not really constants which should be changed. 

A question which may occur: "Why aren't you using a string?" 
Answer:
If i was using a string i would need to use `chr()` to convert the integer value from `read()` into a char. 
But `chr()` throws errors when a invalid int is being converted to a char. 

Because i did not want a except/catch block i simply used the raw bytes instead.

Reply:
Could you please add that with `bytearray` into a comment? Otherwise it looks like some unexplainable magic.